libmcount/mcount.c

```diff
@@ -634,6 +626,9 @@ void mcount_exit_filter_record(struct mcount_thread_data *mtdp,
 
 			if (record_trace_data(mtdp, rstack, retval) < 0)
 				pr_err("error during record");
+		} else {
+			/* invalidate events if filtered-out by time filter */
+			mtdp->nr_events = 0;
```

Comment:
The reason I do this is that SDT events are recorded in a signal handler context.  So it's possible to have multiple events before a function exit and it'll lose some since it can have up to 4 events at once.  So I wanted to flush events as soon as possible regardless of time-filter.

So you need to check whether a event belong to that function actually.  If may be or may be not (it belongs to a parent function).  Also some events (e.g. linux:schedule and kernel events) are not handled this way so we need to filter out them in replay time.

Reply:
Hmm.. I think it's a bit confusing to think read trigger as other events. Why don't we separate read trigger records from other events and put it into `mcount_ret_stack` instead of keeping in `mcount_thread_data` so that it can better support time-filter?

Reply:
Adding another buffer for each rstack might not be a good option due to memory usage.  I'm ok if you could make it to reuse argbuf (when used with -A option).